includes/Services/FlowService.php

```diff
@@ -0,0 +1,199 @@
+<?php
+namespace NewfoldLabs\WP\Module\Onboarding\Services;
+
+use NewfoldLabs\WP\Module\Onboarding\Data\Options;
+use NewfoldLabs\WP\Module\Onboarding\Data\Flows;
+use NewfoldLabs\WP\Module\Onboarding\Data\Data;
+
+class FlowService {
+
+	public static function initalize_flow_data() { 
+        if ( ! ( $flow_data = self::read_flow_data_from_wp_option() ) ) {
+            $flow_data =  self::get_default_flow_data();
+			return self::update_wp_options_data_in_database( $flow_data );
+		}
+		$default_flow_data = Flows::get_data();
+		$updated_flow_data = array();
+		$updated_flow_data = self::update_flow_data_recursive($default_flow_data, $flow_data, $updated_flow_data);
+		return self::update_wp_options_data_in_database($updated_flow_data);
+    }
+
+    public static function get_default_flow_data() {
+		// check if data is available in the database if not then fetch the default dataset
+		$flow_data              = Flows::get_data();
+		$flow_data['createdAt'] = time();
+		// get current flow type
+		$flow_type = Data::current_flow();
+		// update default data if flow type is ecommerce
+		if($flow_type == 'ecommerce') 
+			$flow_data = self::update_default_data_for_ecommerce( $flow_data );
+		self::update_wp_options_data_in_database( $flow_data );
+		return $flow_data;
+	}
+
+	/*
+	 * function to update the Flow Data (Blueprint) in an array recursively in comparison to Flows::get_data() (Database)
+	 */	
+	private static function update_flow_data_recursive(&$default_flow_data, &$flow_data, &$updated_flow_data) {
+		{
+			$flow_data_fixes = Flows::get_fixes();
+
+			foreach ($default_flow_data as $key => $value)
+			{ 
+
+				// Any Key renamed is updated in the database with NewKey and the value from the OldKey is retained
+				if($flow_data_fixes) {
+					foreach ($flow_data_fixes as $old_key=>$new_val) {
+						if (isset($flow_data[$old_key]) && array_key_exists($new_val['new_key'], $default_flow_data)) {
+							if($new_val['retain_existing_value'])
+								$flow_data[$new_val['new_key']] = $flow_data[$old_key];
+							else
+								$flow_data[$new_val['new_key']] = $default_flow_data[$new_val['new_key']];
+							unset($flow_data[$old_key]);
+						}	
+					}
+				}
+
+				// To update an existing value if the key exists in both, the blueprint and database
+				// All keys in the database and not in blueprint are not included
+				if (isset($flow_data) && array_key_exists($key, $flow_data) && !is_array($value)) {
```

Comment:
Looking at the code I can think of a bug, I've tested and recorded a SC so I don't have to type the whole scenario out.
[screen-capture.webm](https://user-images.githubusercontent.com/38878906/206372600-fc21e861-5e05-475e-be14-071580534fba.webm)
Used example of `isFirstNFDOnboarding` but can easily happen for other values as well where we intentionally set it to a negative value like an empty array where default data does not have an empty array but the flow data in the DB does.